docs/proposals/20220712-cluster-api-addon-orchestration.md

```diff
@@ -0,0 +1,437 @@
+
+#### Defining the list of Cluster add-ons to install
+
+HelmChartProxy allows users to define the add-ons to install and which clusters they should be installed on using the `clusterSelector`. The specified Helm chart is then installed on every cluster that includes a label matching the cluster selector and makes it part of the orchestration. If a selected cluster no longer matches the cluster selector label, then the Helm release will be uninstalled. 
+
+Note that the cluster selector label differs from the match label field in ClusterResourceSets as HelmChartProxy specifies a single key value pair whereas the match labels field matches any number of labels. The match labels field was not used because by adding another label pair, all of the currently matched clusters would be unselected and that would trigger the deletion of all associated Helm releases. 
+
+#### Installing and lifecycle managing the package manager
+
+ClusterAddonProvider for Helm imports Helm v3 as a library, making this a no-op. However, it is worth noting that each version of Helm only supports certain version of Kubernetes. For example, the [Helm docs](https://helm.sh/docs/topics/version_skew/#supported-version-skew) mention that Helm 3.9 supports Kubernetes 1.21 through 1.24. As a result, there may need to be multiple versions of HelmChartProxy and HelmReleaseProxy in order to be backwards compatible.
+
+Additionally, other package managers like kapp require in-cluster components such as the kapp-controller. In this case, ClusterAddonProvider should include those in-cluster components as well.
+
+#### Mapping HelmChartProxies to Helm charts
+
+A Helm chart is uniquely identified using the URL of its repository, the name of the chart, and the chart version. As a result, each HelmChartProxy uses the `repoURL`, `chartName`, and `version` fields to uniquely identify a Helm chart and determine which version to install. This allows HelmChartProxy to be compatible with any valid Helm repository and chart.
+
+#### Mapping Kubernetes version to add-ons version
+
+Cluster add-ons by definition have a lifecycle strictly linked to the Cluster lifecycle, and the main consequences of this is the fact that there is a strict correlation between the Cluster’s Kubernetes version and the add-on version to be installed in a Cluster.
+
+In some cases this correlation is explicit, like e.g. for CPI following the “Versioning Policy for External Cloud Providers”; in other cases the compatibility matrix is defined case by case in documentation, like for “Supported version” in the Calico documentation.
+ 
+On top of that, in many Cluster API installations, the add-on version an operator would like to use is limited to a list of certified versions defined in each organization/product vendor.
```

Comment:
@fabriziopandini Just to clarify, does this mean that add-on versions are restricted to the versions that are compatible with their existing Cluster components? Or are there additional cases we want to consider too?

Reply:
AFAIK product vendors usually certify only a small combinations of component versions.
But I don't think that this requires additional work from our side at this stage